### DISCUSSION TITLE
Does Triton support LLaMA2?

### DISCUSSION THREAD

**Discussion by realhaik** [triton-inference-server/server#6212] [Q&A]
Can I run llama-2-7b-chat on Triton? 
Any link to example code will be very helpful. 
https://ai.meta.com/llama/

It must be the original "ckpt" version of the Meta LLaMA2 model that you download from their website: https://ai.meta.com/llama/
I know that Triton supports the Hugging Face version, but unfortunately their model is defective, the results are completely broken.

**Comment by realhaik** [comment]


**Reply by the-david-oy** [reply]
Triton supports a variety of models. It really depends on what framework they are using and if we have a backend that supports it. It looks like their model may be runnable in Python. If so, the Python backend could work, especially given its flexibility. If not, a custom backend would need to be implemented.

**Reply by realhaik** [reply]
Not good... I need a one-liner.

**Reply by the-david-oy** [reply]
Unfortunately, there is not a pre-written guide on deploying this exact model or one that distills that entire process down to one line for you. If you would like to contribute that guide, we would welcome that. 

If you need more hands-on support, please feel free to look into NVIDIA AI Enterprise support (see this link [here](https://www.nvidia.com/en-us/data-center/products/ai-enterprise/)). Good luck with your deployment.

**Reply by yeahdongcn** [reply]
It would be great if someone could add more tutorials at https://github.com/triton-inference-server/tutorials for popular models like Llama 2.

**Reply by the-david-oy** [reply]
Thanks! We are working on it and hope to have some up soon. Any user contributions are also very welcome.

**Reply by yeahdongcn** [reply]
Just noticed https://catalog.ngc.nvidia.com/orgs/nvidia/teams/playground/models/codellama and in [overview section](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/playground/models/codellama/overview) shows it served by Triton. Is it possible to share the best practices in deploying codellama in Triton? Thanks.

```
Inference:
Engine: [Triton](https://developer.nvidia.com/triton-inference-server)
Test Hardware: Other
```

**Reply by realhaik** [reply]
@yeahdongcn 
The nvidia codellama link you posted, is probably runnung the Hugging Face quantized model, or any other quantized model. I can tell that by the broken inference results when I run my tests. The orignial version by meta performs far better. 
My orignal question is how to run the orignal meta model on Triton.

**Comment by realhaik** [comment]
Still no one knows if Nvidia Triton supports LLaMA2.
Everyone knows that "theoretically" it may support it, but no one is able to tell for sure... Very interesting.

**Comment by realhaik** [comment]
> Thanks! We are working on it and hope to have some up soon. Any user contributions are also very welcome.

You are working on it? This should be like a 10 minute task?
